sql/core/src/main/scala/org/apache/spark/sql/execution/metric/CustomMetrics.scala

```diff
@@ -52,7 +55,18 @@ object CustomMetrics {
       currentMetricsValues: Seq[CustomTaskMetric],
       customMetrics: Map[String, SQLMetric]): Unit = {
     currentMetricsValues.foreach { metric =>
-      customMetrics.get(metric.name()).map(_.set(metric.value()))
+      val metricName = metric.name()
+      val metricValue = metric.value()
+      customMetrics.get(metricName).map(_.set(metricValue))
+
+      if (builtInOutputMetrics.contains(metricName)) {
+        Option(TaskContext.get()).map(_.taskMetrics().outputMetrics).foreach { outputMetrics =>
+          metricName match {
+            case "bytesWritten" => outputMetrics.setBytesWritten(metricValue)
+            case "recordsWritten" => outputMetrics.setRecordsWritten(metricValue)
```

Comment:
It will break Scala 2.13. We need to handle all cases exhaustively from Scala 2.13+. Could you add the default case?

Reply:
Yea, I'll add it. Thanks.